Generated Zod throws a TypeScript error

In the latest versions of Zod (3.23.8) and TypeScript (5.4.5) this generated code:
https://github.com/StefanTerdell/json-schema-to-zod/blob/969a0b912f56601e69cebcb2e0b212275be0d191/src/parsers/parseOneOf.ts#L23-L29
throw this TypeScript error:
```
error TS2769: No overload matches this call.
  Overload 1 of 3, '(callbackfn: (previousValue: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<"forever">, currentValue: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<...>, currentIndex: number, array: (ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>)[]) => ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>, initialValue: ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>): ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>', gave the following error.
    Argument of type '(errors: ZodError<any>[], schema: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<"forever">) => (ZodError<any> | undefined)[]' is not assignable to parameter of type '(previousValue: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<"forever">, currentValue: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<...>, currentIndex: number, array: (ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>)[]) => ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>'.
      Types of parameters 'errors' and 'previousValue' are incompatible.
        Type 'ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<"forever">' is not assignable to type 'ZodError<any>[]'.
          Type 'ZodLiteral<"once">' is missing the following properties from type 'ZodError<any>[]': length, pop, push, concat, and 35 more.
  Overload 2 of 3, '(callbackfn: (previousValue: ZodError<any>[], currentValue: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<"forever">, currentIndex: number, array: (ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>)[]) => ZodError<...>[], initialValue: ZodError<...>[]): ZodError<...>[]', gave the following error.
    Argument of type '(errors: ZodError<any>[], schema: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<"forever">) => (ZodError<any> | undefined)[]' is not assignable to parameter of type '(previousValue: ZodError<any>[], currentValue: ZodLiteral<"once"> | ZodLiteral<"repeating"> | ZodLiteral<"forever">, currentIndex: number, array: (ZodLiteral<...> | ... 1 more ... | ZodLiteral<...>)[]) => ZodError<...>[]'.
      Type '(ZodError<any> | undefined)[]' is not assignable to type 'ZodError<any>[]'.
        Type 'ZodError<any> | undefined' is not assignable to type 'ZodError<any>'.
          Type 'undefined' is not assignable to type 'ZodError<any>'.

2802                     (errors: z.ZodError[], schema) =>
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2803                       ((result) =>
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ... 
2805                         schema.safeParse(x),
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2806                       ),
     ~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/package.json
+++ b/package.json
@@ -20,7 +20,7 @@
     "build:types": "tsc -p tsconfig.types.json",
     "build:cjs": "tsc -p tsconfig.cjs.json && node postcjs.js",
     "build:esm": "tsc -p tsconfig.esm.json && node postesm.js",
-    "build": "$ i && npm run gen && npm test && rimraf ./dist && npm run build:types && npm run build:cjs && npm run build:esm",
+    "build": "npm i && npm run gen && npm test && rimraf ./dist && npm run build:types && npm run build:cjs && npm run build:esm",
     "dry": "npm run build && npm pub --dry-run",
     "dev": "tsx watch test/index.ts",
     "gen": "tsx ./createIndex.ts",

--- a/package.json
+++ b/package.json
@@ -20,11 +20,12 @@
     "build:types": "tsc -p tsconfig.types.json",
     "build:cjs": "tsc -p tsconfig.cjs.json && node postcjs.js",
     "build:esm": "tsc -p tsconfig.esm.json && node postesm.js",
-    "build": "npm i && npm run gen && npm test && rimraf ./dist && npm run build:types && npm run build:cjs && npm run build:esm",
+    "build": "$ i && npm run gen && npm test && rimraf ./dist && npm run build:types && npm run build:cjs && npm run build:esm",
     "dry": "npm run build && npm pub --dry-run",
     "dev": "tsx watch test/index.ts",
     "gen": "tsx ./createIndex.ts",
-    "test": "tsx test/index.ts"
+    "test": "tsx test/index.ts",
+    "prepare": "$npm_execpath run gen && $npm_execpath run test && rimraf ./dist && $npm_execpath run build:types && $npm_execpath run build:cjs && $npm_execpath run build:esm"
   },
   "keywords": [
     "zod",

--- a/src/parsers/parseOneOf.ts
+++ b/src/parsers/parseOneOf.ts
@@ -23,7 +23,7 @@ export const parseOneOf = (
     const errors = schemas.reduce<z.ZodError[]>(
       (errors, schema) =>
         ((result) =>
-          "error" in result && typeof result.error !== 'undefined' ? [...errors, result.error] : errors)(
+          result.error ? [...errors, result.error] : errors)(
           schema.safeParse(x),
         ),
       [],

--- a/src/parsers/parseOneOf.ts
+++ b/src/parsers/parseOneOf.ts
@@ -20,12 +20,13 @@ export const parseOneOf = (
         }),
       )
       .join(", ")}];
-    const errors = schemas.reduce(
-      (errors: z.ZodError[], schema) =>
-        ((result) => ("error" in result ? [...errors, result.error] : errors))(
-          schema.safeParse(x)
+    const errors = schemas.reduce<z.ZodError[]>(
+      (errors, schema) =>
+        ((result) =>
+          "error" in result && typeof result.error !== 'undefined' ? [...errors, result.error] : errors)(
+          schema.safeParse(x),
         ),
-      []
+      [],
     );
     if (schemas.length - errors.length !== 1) {
       ctx.addIssue({

--- a/test/parsers/parseObject.test.ts
+++ b/test/parsers/parseObject.test.ts
@@ -212,12 +212,13 @@ suite("parseObject", (test) => {
 
       `z.object({ "a": z.string() }).and(z.any().superRefine((x, ctx) => {
     const schemas = [z.object({ "b": z.string() }), z.object({ "c": z.string() })];
-    const errors = schemas.reduce(
-      (errors: z.ZodError[], schema) =>
-        ((result) => ("error" in result ? [...errors, result.error] : errors))(
-          schema.safeParse(x)
+    const errors = schemas.reduce<z.ZodError[]>(
+      (errors, schema) =>
+        ((result) =>
+          "error" in result && typeof result.error !== 'undefined' ? [...errors, result.error] : errors)(
+          schema.safeParse(x),
         ),
-      []
+      [],
     );
     if (schemas.length - errors.length !== 1) {
       ctx.addIssue({

--- a/test/parsers/parseObject.test.ts
+++ b/test/parsers/parseObject.test.ts
@@ -215,7 +215,7 @@ suite("parseObject", (test) => {
     const errors = schemas.reduce<z.ZodError[]>(
       (errors, schema) =>
         ((result) =>
-          "error" in result && typeof result.error !== 'undefined' ? [...errors, result.error] : errors)(
+          result.error ? [...errors, result.error] : errors)(
           schema.safeParse(x),
         ),
       [],

--- a/test/parsers/parseOneOf.test.ts
+++ b/test/parsers/parseOneOf.test.ts
@@ -17,12 +17,13 @@ suite("parseOneOf", (test) => {
       ),
       `z.any().superRefine((x, ctx) => {
     const schemas = [z.string(), z.number()];
-    const errors = schemas.reduce(
-      (errors: z.ZodError[], schema) =>
-        ((result) => ("error" in result ? [...errors, result.error] : errors))(
-          schema.safeParse(x)
+    const errors = schemas.reduce<z.ZodError[]>(
+      (errors, schema) =>
+        ((result) =>
+          "error" in result && typeof result.error !== 'undefined' ? [...errors, result.error] : errors)(
+          schema.safeParse(x),
         ),
-      []
+      [],
     );
     if (schemas.length - errors.length !== 1) {
       ctx.addIssue({

--- a/test/parsers/parseOneOf.test.ts
+++ b/test/parsers/parseOneOf.test.ts
@@ -20,7 +20,7 @@ suite("parseOneOf", (test) => {
     const errors = schemas.reduce<z.ZodError[]>(
       (errors, schema) =>
         ((result) =>
-          "error" in result && typeof result.error !== 'undefined' ? [...errors, result.error] : errors)(
+          result.error ? [...errors, result.error] : errors)(
           schema.safeParse(x),
         ),
       [],

